openrtx/include/core/user_functions.h

```diff
@@ -0,0 +1,60 @@
+/***************************************************************************
+ *   Copyright (C) 2020 - 2024 by Federico Amedeo Izzo IU2NUO,             *
+ *                                Niccolò Izzo IU2KIN,                     *
+ *                                Silvano Seva IU2KWO                      *
+ *                                Morgan Diepart ON4MOD                    *
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 3 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ *   This program is distributed in the hope that it will be useful,       *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU General Public License for more details.                          *
+ *                                                                         *
+ *   You should have received a copy of the GNU General Public License     *
+ *   along with this program; if not, see <http://www.gnu.org/licenses/>   *
+ ***************************************************************************/
+
+#ifndef USER_FUNCTIONS_H
+#define USER_FUNCTIONS_H
+
+#include <hwconfig.h>
+#include <stdint.h>
+#include <stddef.h>
+#include <errno.h>
+
+#if defined(CONFIG_USER_FUNCTIONS) && (CONFIG_USER_FUNCTIONS > 0)
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+
+typedef enum {
+    USER_FUNCTION_ASYNC = 0,
+    USER_FUNCTION_100HZ = 10,           // 10 ticks
+    USER_FUNCTION_50HZ  = 20,           // 20 ticks
+    USER_FUNCTION_20HZ  = 50,           // 50 ticks
+    USER_FUNCTION_10HZ  = 100,          // 100 ticks
+    USER_FUNCTION_1HZ   = 1000,         // 1000 ticks
+} user_functions_sched_t;
+
+void user_functions_init();
```

Comment:
Please add doxygen-style documentation to the functions. :)

Reply:
Will do